### PR TITLE
Refresh token & Refactoring

### PR DESCRIPTION
**Pull request by thatsddr**
### What's new

- Added api routes for login, logout, and token refresh
- API tests

### What changed

- Login tests
- Auth logic has been moved
- Refactored forms
- Refactored navbar
- Refactored API
- Refactored contexts
- Fixed bugs in the landing page

### How to test

> I know that there is a lot of new code by running the tests and running this locally should make you confident that this works how it should. Much of the "new" code is old code refactored.

- Download/pull the code
- Run ```git checkout refresh_token```
- Run ```npm i```
- Run ```npm run test```
- [Optional] run ```npm run dev``` if you want to run this locally

### CODE REVIEW

**Comment by Darjusch** [comment]
Nextjs
I got an error running the tests

File: __tests__/api/commonFunctions.test.ts [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+import mock from "mock-http";
+import { NextApiRequest } from "next";
+
+describe("Test the functions that make the proxy work", () => {
+	let req: NextApiRequest,
+		res: any,
+		rej: any,
+		resolve: any,
+		token =
+			"eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJleHAiOjE5OTk5OTk5OTksImVtYWlsIjoidGVzdEB0ZXN0LmNvIiwiaWF0IjoxNTE2MjM5MDIyfQ.JyrM7CQymsHChoqFcj_-VCJwn0mDQtN9r8jnEJ_TySw";
```
Comment (Darjusch) [inline __tests__/api/commonFunctions.test.ts:18]: is this a valid one ?

Comment (thatsddr) [reply __tests__/api/commonFunctions.test.ts:18]: No, it's one I generated with jwt.io. Actually, all the tokens that you can find in the test are generated with that tool

**Comment by thatsddr** [comment]
I will fix the error asap, it happens because we no longer redirect the users to the login form but we expect them to verify their emails first

**Comment by thatsddr** [comment]
Now it should work😀

**Pull request merged**